brew install gmp fail (show leopard)

==> ./configure --disable-static --prefix=/usr/local/Cellar/gmp/6.2.1 --enable-cxx
==> make
Last 15 lines from /Users/palo/Library/Logs/Homebrew/gmp/02.make:
libtool: compile:  /usr/bin/gcc-4.2 -std=gnu99 -DHAVE_CONFIG_H -I. -I.. -D__GMP_WITHIN_GMP -I.. -DOPERATION_scan1 -Os -w -pipe -march=core2 -mmacosx-version-min=10.6 -c scan1.c  -fno-common -DPIC -o .libs/scan1.o
/bin/sh ../libtool --mode=compile --tag=CC ../mpn/m4-ccas --m4="m4" /usr/bin/gcc-4.2 -std=gnu99 -c -DHAVE_CONFIG_H -I. -I..  -D__GMP_WITHIN_GMP -I.. -DOPERATION_`echo hamdist | sed 's/_$//'`   -Os -w -pipe -march=core2 -mmacosx-version-min=10.6  `test -f 'hamdist.asm' || echo './'`hamdist.asm
libtool: compile:  ../mpn/m4-ccas --m4=m4 /usr/bin/gcc-4.2 -std=gnu99 -c -DHAVE_CONFIG_H -I. -I.. -D__GMP_WITHIN_GMP -I.. -DOPERATION_popcount -Os -w -pipe -march=core2 -mmacosx-version-min=10.6 popcount.asm  -fno-common -DPIC -o .libs/popcount.o
m4  -DHAVE_CONFIG_H -D__GMP_WITHIN_GMP -DOPERATION_popcount -DPIC popcount.asm >tmp-popcount.s
 /usr/bin/gcc-4.2 -std=gnu99 -c -DHAVE_CONFIG_H -I. -I.. -D__GMP_WITHIN_GMP -I.. -DOPERATION_popcount -Os -w -pipe -march=core2 -mmacosx-version-min=10.6 tmp-popcount.s -fno-common -DPIC -o .libs/popcount.o
tmp-popcount.s:181:suffix or operands invalid for `movq'
make[2]: *** [popcount.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
libtool: compile:  ../mpn/m4-ccas --m4=m4 /usr/bin/gcc-4.2 -std=gnu99 -c -DHAVE_CONFIG_H -I. -I.. -D__GMP_WITHIN_GMP -I.. -DOPERATION_hamdist -Os -w -pipe -march=core2 -mmacosx-version-min=10.6 hamdist.asm  -fno-common -DPIC -o .libs/hamdist.o
m4  -DHAVE_CONFIG_H -D__GMP_WITHIN_GMP -DOPERATION_hamdist -DPIC hamdist.asm >tmp-hamdist.s
 /usr/bin/gcc-4.2 -std=gnu99 -c -DHAVE_CONFIG_H -I. -I.. -D__GMP_WITHIN_GMP -I.. -DOPERATION_hamdist -Os -w -pipe -march=core2 -mmacosx-version-min=10.6 tmp-hamdist.s -fno-common -DPIC -o .libs/hamdist.o
tmp-hamdist.s:206:suffix or operands invalid for `movq'
make[2]: *** [hamdist.lo] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2

--- a/Library/Formula/gmp.rb
+++ b/Library/Formula/gmp.rb
@@ -9,6 +9,11 @@ class Gmp < Formula
     sha256 "8959b62e4b35331839445da5d5047bde738d7a53ab0abfe08f24249f135c3312" => :tiger_altivec
   end
 
+  # "suffix or operands invalid for movq"
+  # https://www.mail-archive.com/gmp-bugs@gmplib.org/msg01321.html
+  # https://gmplib.org/repo/gmp/rev/3ac5afa36be5
+  patch :DATA
+
   option "32-bit"
   option :cxx11
 
@@ -71,3 +76,48 @@ class Gmp < Formula
     system "./test"
   end
 end
+__END__
+--- a/mpn/x86_64/bd1/hamdist.asm	Fri Oct 28 16:31:44 2022 +0200
++++ b/mpn/x86_64/bd1/hamdist.asm	Wed Nov 02 13:48:37 2022 +0100
+@@ -170,7 +170,7 @@
+ 	paddq	%xmm0, %xmm8
+ 	pshufd	$14, %xmm8, %xmm0
+ 	paddq	%xmm8, %xmm0
+-	movq	%xmm0, %rax
++	movd	%xmm0, %rax
+ 	add	%r10, %rax
+ 	FUNC_EXIT()
+ 	ret
+--- a/mpn/x86_64/bd1/popcount.asm	Fri Oct 28 16:31:44 2022 +0200
++++ b/mpn/x86_64/bd1/popcount.asm	Wed Nov 02 13:48:37 2022 +0100
+@@ -167,7 +167,7 @@
+ 	paddq	%xmm5, %xmm8
+ 	pshufd	$14, %xmm8, %xmm0
+ 	paddq	%xmm8, %xmm0
+-	movq	%xmm0, %rax
++	movd	%xmm0, %rax
+ 	add	%rdx, %rax
+ 	FUNC_EXIT()
+ 	ret
+--- a/mpn/x86_64/core2/hamdist.asm	Fri Oct 28 16:31:44 2022 +0200
++++ b/mpn/x86_64/core2/hamdist.asm	Wed Nov 02 13:48:37 2022 +0100
+@@ -191,7 +191,7 @@
+ 	paddq	%xmm4, %xmm8
+ 	pshufd	$14, %xmm8, %xmm0
+ 	paddq	%xmm8, %xmm0
+-	movq	%xmm0, %rax
++	movd	%xmm0, %rax
+ 	ret
+ EPILOGUE()
+ DEF_OBJECT(L(cnsts),16,`JUMPTABSECT')
+--- a/mpn/x86_64/core2/popcount.asm	Fri Oct 28 16:31:44 2022 +0200
++++ b/mpn/x86_64/core2/popcount.asm	Wed Nov 02 13:48:37 2022 +0100
+@@ -166,7 +166,7 @@
+ 	paddq	%xmm4, %xmm8
+ 	pshufd	$14, %xmm8, %xmm0
+ 	paddq	%xmm8, %xmm0
+-	movq	%xmm0, %rax
++	movd	%xmm0, %rax
+ 	ret
+ EPILOGUE()
+ DEF_OBJECT(L(cnsts),16,`JUMPTABSECT')

